onnxruntime/core/providers/cpu/controlflow/if.cc

```diff
@@ -90,15 +91,56 @@ class IfImpl {
   std::vector<std::pair<AllocationType, MLValue>> outputs_;
 };
 
+void If::CheckForPassthroughOptimization(const OpKernelInfo& info,
+                                         const ONNX_NAMESPACE::GraphProto& then_proto,
+                                         const ONNX_NAMESPACE::GraphProto& else_proto) {
+  // check if subgraph has a single identity node passing a single input through as a single output
+  auto check_identity_passthrough = [](const ONNX_NAMESPACE::GraphProto& proto) -> const std::string {
+    if (proto.node_size() == 1 && proto.output_size() == 1) {
+      const auto& node = proto.node().Get(0);
+      if (node.op_type() == "Identity") {
+        const auto& output_name = node.output().Get(0);
+        const auto& graph_output_name = proto.output().Get(0).name();
+
+        // if the Identity node is producing the graph output, we can just copy the input directly in If::Compute
+        if (graph_output_name == output_name) {
```

Comment:
in the case "graph_output_name != output_name", pass thru is not good? why?

Reply:
this test should really never fail so was just a paranoid check. we already established that the graph had a single Identity node, and a single output, so I don't know where else you would get a different graph output from. but just in case that was possible, I added a check.  

if the identity node wasn't producing the output, I don't know what the correct input to copy would be, so in that case I don't know what to pass through.